### DISCUSSION TITLE
Cannot connect to GCP standalone kubeflow pipeline

### DISCUSSION THREAD

**Discussion by superjcd** [elyra-ai/elyra#3045] [Pipeline Editor/ Kubeflow Pipelines + Apache Airflow]
I used the GCP standalone kubeflow pipeline as my pipeline runtime  with `No Authneitcation`(only set the Kubeflow Pipelines API Endpoint), but here i got error like:

![image](https://user-images.githubusercontent.com/25010312/206101810-893ec6d2-c802-4524-adee-cd97d204b29a.png)

Can anyone helps me?

**Comment by ptitzler** [comment]
Hi @superjcd,

what happens when you click on the `Kubeflow Pipelines API endpoint` link for your runtime configuration? If the Kubeflow Pipelines Central dashboard doesn't open, then the URL you've entered is likely invalid.

![image](https://user-images.githubusercontent.com/13068832/206230746-fd130fcb-c1e4-40e6-aa29-1747c7f870b3.png)

**Reply by superjcd** [reply]
The dashborad opened correctly:
![image](https://user-images.githubusercontent.com/25010312/206332442-267974a2-00ce-43bc-8bff-3fc1b7bb4b01.png)
I guess it's something to do with the GCP access token?

**Reply by ptitzler** [reply]
I don't have access to GCP and therefore can't really troubleshoot this. Based on the information in https://cloud.google.com/ai-platform/pipelines/docs/connecting-with-sdk, this might be a problem with the set up. Can you confirm whether your environment meets the prerequisites listed in the _Before you start_ section. To determine whether this is an Elyra-specific issue, you can write a small Python program that attempts to connect to your KFP endpoint and run it in the same environment where Elyra is deployed:

```
import kfp
client = kfp.Client(host='https://...pipelines.googleusercontent.com...')
client.list_pipelines()
```

If that works it could be an issue with our [dummy authentication implementation](https://github.com/elyra-ai/elyra/blob/main/elyra/pipeline/kfp/kfp_authentication.py#L348). To confirm if that's the case, you can run this code

```
import requests
from urllib.parse import urlsplit

url = urlsplit("https://...pipelines.googleusercontent.com...")._replace(path="").geturl()
requests.get(url, allow_redirects=True)
```

**Reply by superjcd** [reply]
I think I've walked through the prerequisites before, but the `kfp` sdk still throw me a warning and i cant't get any of my pipelines:
![image](https://user-images.githubusercontent.com/25010312/206606589-5105389f-c67b-442d-a947-efaf5303b6a9.png)
But interesting thing is when I add a `existing_token` (which I got via `gcloud auth print-acess-token` on cloud shell) to the `kfp.Client`, then it works perfectly(my windows terminal, not cloud shell):
![image](https://user-images.githubusercontent.com/25010312/206606861-265881c3-b3b6-40ab-9f0b-b24c2450e10f.png)
So I do believe it's related to the GCP access token, but I don't know how to set  on Elyra

## PS
Some related issuses:
- https://github.com/kubeflow/pipelines/issues/2954
- https://github.com/kubeflow/pipelines/issues/4143

**Reply by ptitzler** [reply]
This is very useful information, Thank you!

As a stop gap it appears that we could add a new [authentication provider](https://github.com/elyra-ai/elyra/blob/main/elyra/pipeline/kfp/kfp_authentication.py) that takes as input an existing token. Such an approach would be vendor agnostic, because Elyra doesn't need to implement the process of obtaining the token via the vendor's proprietary authentication flow. It's not very user friendly though, because the token cannot be automatically obtained or refreshed and it's up to the user to manage it.
